src/Generic/Random/Tutorial.hs

```diff
@@ -224,11 +226,56 @@
 --   arbitrary = 'genericArbitrarySingleG' customGens
 -- @
 --
+-- == Example 2 ('ConstrGen')
+--
+-- Here's the @Tree@ type from the beginning again.
+--
+-- @
+-- data Tree a = Leaf a | Node (Tree a) (Tree a)
+--   deriving 'GHC.Generics.Generic'
+-- @
+--
+-- We can generate "linear trees" by forcing every right child of a @Node@ to be a @Leaf@:
+--
+-- @
+-- {-\# LANGUAGE ScopedTypeVariables \#-}
+--
+-- instance Arbitrary a => Arbitrary (Tree a) where
+--   arbitrary = 'genericArbitraryUG' customGens
+--     where
+--       customGens :: 'ConstrGen' \"Node\" 1 (Tree a)    ':+' ()
+--       customGens =  'ConstrGen' (Leaf '<$>' arbitrary) ':+' ()
+--       -- Generator for the right field of constructor Node
+--       -- (fields are indexed from 0), which must have type (Tree a).
+-- @
+--
+-- That instance is equivalent to the following:
+--
+-- @
+-- instance Arbitrary a => Arbitrary (Tree a) where
+--   arbitrary = oneof
+--     [ Leaf '<$>' arbitrary
+--     , Node '<$>' arbitrary '<*>' (Leaf '<$>' arbitrary)
+--     --         ^ recursive call
+--     ]
+-- @
+--
+-- So generated trees look like this ("left-biased linear trees"):
+--
+-- > Node (Node (Node (Node (Leaf 1)
+-- >                        (Leaf 2))
+-- >                  (Leaf 3))
+-- >            (Leaf 4))
+-- >      (Leaf 5)
```

Comment:
I believe a right-biased tree might be easier to parse visually, but it's a great example anyway. Thanks a lot! :)

Reply:
Indeed! I wanted an example where the field index was not 0 but now I'm thinking I do want a 0 there, so I'll change this example...